File: mobile/src/main/java/treehou/se/habit/module/ViewModule.java
package treehou.se.habit.module;


import dagger.Module;
import dagger.Provides;

@Module
public abstract class ViewModule<T> {
    protected final T view;

    public ViewModule(T view) {
        this.view = view;
    }

    @Provides
    @ActivityScope
    public T provideActivity() {
        return view;
    }
}

File: mobile/src/main/java/treehou/se/habit/dagger/ViewModule.kt
package treehou.se.habit.dagger


import dagger.Module
import dagger.Provides
import treehou.se.habit.dagger.scopes.ActivityScope

@Module
abstract class ViewModule<out T>(protected val view: T) {

    @Provides
    @ActivityScope
    fun provideActivity(): T {
        return view
    }
}